test/examples.prl

```diff
@@ -192,31 +192,32 @@ Thm Not : [(-> [_ : wbool] wbool)] by [
   {lam x. if x then `ff else `tt}; auto
 ].
 
-Thm FunExt(#A; #B) : [
-  type/A : #A type,
-  type/B : #B type
-  >>
+Thm FunExt{l:lvl} : [
   (->
-   [f : (-> #A #B)]
-   [g : (-> #A #B)]
-   [p : (-> [y : #A] (path {_} #B ($ f y) ($ g y)))]
-   (path {_} (-> #A #B) f g))
+   [a b : (univ l)]
+   [f g : (-> a b)]
+   [p : (-> [y : a] (path {_} b ($ f y) ($ g y)))]
+   (path {_} (-> a b) f g))
 ] by [
-  lam f, g, p.
-    <i> lam a. use p [use a, @i]
+  lam a, b, f, g, p.
+    <i> lam x. use p [use x, @i]
 ].
 
 Print FunExt.
 
-
-Tac FunExtTac(#A; #B; #F; #G) = [
-  let p : [(-> [x : #A] (path {_} wbool ($ #F x) ($ #G x)))] = id.
-  <i> lam a. use p [use a, @i]
+Tac FunExtTac{l:lvl} = [
+  query gl <- goal.
+  match [gl:jdg] {
+    [a, b, f, g | #jdg{(path {_} (-> %a %b) %f %g) true} =>
+      use {FunExt l}
+        [ `%a, `%b, `%f, `%g, id ]]
+  }
```

Comment:
Here's an example of something that would be cool to generalize and have as a built-in tactic, rather than having to write it manually in each case.

We're using unification to fill in the arguments to the lemma.

Reply:
Do you mean something like `use {FunExt l} [_, _, _, _, id]`? That is, unification everywhere?

Reply:
Right. I think that's possible to implement, but the code will be a little tricky.